### PR TITLE
Backup to open correctly from insights dialog

### LINKED ISSUE #5401: backup button in the backup insights dialog not working properly
the backup button does nothing when the insights dialog is opened second 
Steps to Reproduce:
1. Start (restart) ADS
2. connect to a server and open dashboard
3. open the backup insights dialog (show details) via the "..." of backup status widget
4. click the backup button on the bottom left of the dialog
5. the backup dialog will be opened, close it and close the backup insights dialog
6. open the backup insights dialog (same as step 3)
7. click the backup button

expected:
backup dialog opens

actual:
nothing happens

### PR DESCRIPTION
**Pull request by udeeshagautam**
Fixes #5401 

Insights dialog was reusing custom buttons added in first render (since modal html element preserves the footer buttons) but the OnDidClick function for these returned buttons wasn't coming correctly from previous iteration - removing and recreating footer button solves this issues.

backup from insights

### CODE REVIEW

File: src/sql/workbench/browser/modal/modal.ts [hunk 1]
```diff
@@ -412,6 +412,24 @@ export abstract class Modal extends Disposable implements IThemable {
 		});
 	}
 
+	/**
+	* removes the footer button matching the provided label
+	* @param label Label on the button
+	*/
+	protected removeFooterButton(label: string): void {
+		let buttonIndex = this._footerButtons.findIndex(e => {
+			try {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modal/modal.ts]: Curious why the try/catch is necessary - I wouldn't expect this to every throw.

File: src/sql/workbench/browser/modal/modal.ts [hunk 2]
```diff
@@ -412,6 +412,24 @@ export abstract class Modal extends Disposable implements IThemable {
 		});
 	}
 
+	/**
+	* removes the footer button matching the provided label
+	* @param label Label on the button
+	*/
+	protected removeFooterButton(label: string): void {
+		let buttonIndex = this._footerButtons.findIndex(e => {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modal/modal.ts]: Can just use find to return the item directly - slightly more succient

File: src/sql/workbench/browser/modal/modal.ts [hunk 3]
```diff
@@ -412,6 +412,24 @@ export abstract class Modal extends Disposable implements IThemable {
...
+			try {
+				return e && e.element.innerText === label;
+			} catch {
+				return false;
+			}
+		});
+		let button = this._footerButtons[buttonIndex];
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modal/modal.ts]: Consider adding a check for findIndex not finding the item (buttonIndex === -1) (or undefined if you switch to using find directly) that skips the removal if we couldn't find the item

File: src/sql/workbench/services/insights/browser/insightsDialogView.ts [hunk 4]
```diff
@@ -337,7 +337,11 @@ export class InsightsDialogView extends Modal {
 				let task = tasks.includes(action);
 				let commandAction = MenuRegistry.getCommand(action);
 				let commandLabel = types.isString(commandAction.title) ? commandAction.title : commandAction.title.value;
-				if (task && !this.findFooterButton(commandLabel)) {
+				if (task) {
+					// need to remove and add fresh because the onDidClick action from previous addition is not called
+					if (this.findFooterButton(commandLabel)) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/insights/browser/insightsDialogView.ts]: Oh there's already this method for finding the button. I'd suggest either having removeFooterButton do the check as well and just take in the button object directly or remove the findFooterButton call here and just add it to removeFooterButton

File: src/sql/workbench/browser/modal/modal.ts (hunk 2 above)
Comment (udeeshagautam) [reply src/sql/workbench/browser/modal/modal.ts]: using the index to splice

File: src/sql/workbench/browser/modal/modal.ts (hunk 1 above)
Comment (udeeshagautam) [reply src/sql/workbench/browser/modal/modal.ts]: copied the same patter as find... e.element might throw in worst case

File: src/sql/workbench/browser/modal/modal.ts (hunk 3 above)
Comment (udeeshagautam) [reply src/sql/workbench/browser/modal/modal.ts]: ah yes good catch.. will do

File: src/sql/workbench/services/insights/browser/insightsDialogView.ts (hunk 4 above)
Comment (udeeshagautam) [reply src/sql/workbench/services/insights/browser/insightsDialogView.ts]: This increases readability (to get a sense of what is happening exactly) so want to keep it this way

File: src/sql/workbench/browser/modal/modal.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/browser/modal/modal.ts]: Why not just add to the check below if you're concerned about e.element being undefined?

```
return e && e.element && e.element.innerText
```

It just seems unnecessarily verbose.

File: src/sql/workbench/services/insights/browser/insightsDialogView.ts (hunk 4 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/services/insights/browser/insightsDialogView.ts]: Ok, but you shouldn't be duplicating the find logic. Feel free to keep it split into two parts but this is just making it take longer than it needs to.

Comment (udeeshagautam) [reply src/sql/workbench/services/insights/browser/insightsDialogView.ts]: hmm.... I don't necessarily worry with this less cost. But I am fine with changing to single call too... will remove the find call as you suggested :)

**Pull request merged**